packages/venia-concept/src/index.js

```diff
@@ -39,14 +40,38 @@ const authLink = setContext((_, { headers }) => {
 });
 
 // https://www.apollographql.com/docs/link/links/error/
-const errorLink = onError(({ graphQLErrors, networkError }) => {
+const errorLink = onError(({ graphQLErrors, networkError, response }) => {
     if (graphQLErrors)
         graphQLErrors.forEach(({ message, locations, path }) =>
             console.log(
                 `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
             )
         );
     if (networkError) console.log(`[Network error]: ${networkError}`);
+
+    const { data, errors } = response;
+
+    errors.forEach(({ message, path }, index) => {
+        // It's within the GraphQL spec to receive data and errors, where errors are merely informational and not
+        // intended to block. Almost all existing components were not built with this in mind, so we build special
+        // handling of this error message so we can deal with it at the time we deem appropriate.
+        if (message === 'Some of the products are out of stock.') {
+            const pathToCartItems = path.slice(0, -1);
+            const cartItems = getWithPath(data, pathToCartItems);
+
+            // Until MC-36092 is resolved, we need to guard against a null product being returned as well.
+            if (cartItems[0] === null) {
+                cartItems.splice(0, 1);
+            }
+
+            // Suppress entirely if this is our only error, we don't want to cause operations to throw.
+            if (response.errors.length === 1) {
+                response.errors = undefined;
+            } else {
+                response.errors[index] = undefined;
+            }
```

Comment:
If I'm reading this correctly it looks like we now delete errors from responses. What affect does this have on components that were previously handling these errors? Do the invokers still throw? Does this just prevent `error` from being a property on the response ie `const { error } = useQuery(...)`?

Reply:
Both! `response.errors.length > 0` will cause invokers to throw and the population of `error` in responses. PM wants to emulate Luma in this scope, where we don't force resolution of the error until necessary (checkout). There is an Epic planned where we might do a more automated approach to just auto-remove and pop a toast, but keeping it simple for now.

Reply:
So because we are not slicing it the error still gets populated?